test/e2e/workload_test.go

```diff
@@ -616,3 +654,224 @@ var _ = Describe("Statefulset", func() {
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for sidecar")
+			f.EventuallyDaemonSet(daemonset.ObjectMeta).Should(matcher.HaveSidecar(util.StashContainer))
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs, err := f.GetBackupSession(backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&daemonset).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+			err = framework.WaitUntilBackupConfigurationDeleted(f.StashClient, backupCfg.ObjectMeta)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("Waiting for sidecar to be removed")
+			f.EventuallyDaemonSet(daemonset.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
+			err = f.WaitUntilDaemonPodReady(daemonset.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Deleting sample data from pod")
+			err = f.CleanupSampleDataFromWorkload(daemonset.ObjectMeta, apis.KindDaemonSet)
+			Expect(err).NotTo(HaveOccurred())
+
+		}
+	)
+	Context("General Backup new DaemonSet", func() {
+		BeforeEach(func() {
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			daemonset = f.DaemonSet(pvc.Name)
+			targetref = v1beta1.TargetRef{
+				Name:       daemonset.Name,
+				Kind:       apis.KindDaemonSet,
+				APIVersion: "apps/v1",
+			}
+		})
+		AfterEach(func() {
+			err := f.DeleteDaemonSet(daemonset.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitUntilDaemonSetDeleted(f.KubeClient, daemonset.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			f.DeleteRepositories(f.DaemonSetRepos(&daemonset))
+			err = framework.WaitUntilRepositoriesDeleted(f.StashClient, f.DaemonSetRepos(&daemonset))
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.DeleteRestoreSession(restoreSession.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitUntilRestoreSessionDeleted(f.StashClient, restoreSession.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("General Backup new DaemonSet", func() {
+			By("Creating DaemonSet Backup")
+			testDaemonSetBackup()
+
+			By("Creating Restore Session")
+			err = f.CreateRestoreSession(restoreSession)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilDaemonSetReady(f.KubeClient, daemonset.ObjectMeta)
```

Comment:
What is the use of `util.WaitUntilDaemonSetReady()` here? We don't need it. DaemonSet will go to ready state when `init-container` completes it's restore process and daemon container start successfully. So we should check for daemon readiness after checking those.